cmd/cluster/aws/create.go

```diff
@@ -77,10 +85,84 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 	if err != nil {
 		return err
 	}
-	infraID := opts.InfraID
-
 	// Load or create infrastructure for the cluster
 	var infra *awsinfra.CreateInfraOutput
+	var iamInfo *awsinfra.CreateIAMOutput
+	infraID := opts.InfraID
+
+	if infraID == "" {
+		return ErrMsg("infra-id")
+	}
+	hci := hyperv1.HostedClusterInfrastructure{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      opts.InfraID,
+			Namespace: opts.Namespace,
+		},
+	}
+	if err := client.Get(ctx, crclient.ObjectKeyFromObject(&hci), &hci); err != nil {
+		if apierrors.IsNotFound(err) {
+			opts.Log.Info("Hosted cluster infrastructure not found, creating infrastructure with user input", "namespace", opts.Namespace, "name", opts.InfraID)
+		} else {
+			return err
+		}
+	} else {
+		opts.Log.Info("Found hosted cluster infrastructure", "namespace", opts.Namespace, "name", opts.InfraID)
+		if hci.Spec.SubDomain != opts.Name {
+			return fmt.Errorf(fmt.Sprintf("The HostedClusterInfrastructure.Spec.SubDomain must equal the cluster name\nCluster name: %s, Sub-Domain: %s", opts.Name, hci.Spec.SubDomain))
+		}
+		//Set a finalizer on HCI
+		origHCI := hci.DeepCopy()
+		controllerutil.AddFinalizer(&hci, core.DestroyClusterFinalizer)
+		if hci.ObjectMeta.Labels == nil {
+			hci.SetLabels(map[string]string{})
+		}
+		hci.ObjectMeta.Labels[hostedClusterLabel] = opts.Name
+		if err := client.Patch(ctx, &hci, crclient.MergeFrom(origHCI)); err != nil {
+			return err
+		}
+
+		if opts.ServiceCIDR == "" {
+			exampleOptions.ServiceCIDR = "172.31.0.0/16"
+		}
+		if opts.ClusterCIDR == "" {
+			exampleOptions.ClusterCIDR = "10.132.0.0/14"
+		}
+		if opts.ControlPlaneAvailabilityPolicy == "" {
+			exampleOptions.ControlPlaneAvailabilityPolicy = "SingleReplica"
+		}
+		infra = &awsinfra.CreateInfraOutput{
+			Name:            opts.Name,
+			Region:          hci.Spec.Platform.AWS.Region,
+			BaseDomain:      hci.Spec.BaseDomain,
+			InfraID:         hci.Spec.InfraID,
+			MachineCIDR:     hci.Status.Infrastructure.Networking.MachineNetwork[0].CIDR.String(),
+			VPCID:           hci.Status.Infrastructure.Platform.AWS.VPC,
+			SecurityGroupID: *hci.Status.Infrastructure.Platform.AWS.SecurityGroups[0].ID,
+			PublicZoneID:    hci.Status.Infrastructure.DNS.PublicZoneID,
+			PrivateZoneID:   hci.Status.Infrastructure.DNS.PrivateZoneID,
```

Comment:
do we need to check to make sure the HCI is completed before starting to use it? I don't see any check that guarantees the HCI is done

Reply:
Fixed.
```
$ hy create cluster aws --name jnp-test3 --infra-id jnp-infra3 --pull-secret ~/pull-secret --node-pool-replicas 2
2022-10-18T22:54:58-04:00       INFO    Found hosted cluster infrastructure     {"namespace": "clusters", "name": "jnp-infra3"}
2022-10-18T22:54:58-04:00       ERROR   Failed to create cluster        {"error": "the hosted cluster infrastructure resource is not configured as expected\n- provider-secret  : false\n- infra-as-expected: false\n- iam-as-expected  : false"}
github.com/spf13/cobra.(*Command).execute
        /home/jpacker/workspace_git/hypershift/vendor/github.com/spf13/cobra/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        /home/jpacker/workspace_git/hypershift/vendor/github.com/spf13/cobra/command.go:974
github.com/spf13/cobra.(*Command).Execute
        /home/jpacker/workspace_git/hypershift/vendor/github.com/spf13/cobra/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext
        /home/jpacker/workspace_git/hypershift/vendor/github.com/spf13/cobra/command.go:895
main.main
        /home/jpacker/workspace_git/hypershift/main.go:70
runtime.main
        /usr/local/go/src/runtime/proc.go:250
Error: the hosted cluster infrastructure resource is not configured as expected
- provider-secret  : false
- infra-as-expected: false
- iam-as-expected  : false
the hosted cluster infrastructure resource is not configured as expected
- provider-secret  : false
- infra-as-expected: false
- iam-as-expected  : false
```